// @progress/kendo-react-conversational-ui/dist/es/components/NewMessage.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
import { Keys, classNames } from '@progress/kendo-react-common';
import * as React from 'react';
import Toolbar from './Toolbar';
import { Button } from '@progress/kendo-react-buttons';
import { paperPlaneIcon, moreHorizontalIcon } from '@progress/kendo-svg-icons';
import { provideLocalizationService } from '@progress/kendo-react-intl';
import { messages, newMessageSendButton, newMessageOptionsButton, newMessageMessageInput } from '../messages';
/**
 * @hidden
 */
var NewMessage = /** @class */ (function (_super) {
    __extends(NewMessage, _super);
    function NewMessage() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.state = { focused: false };
        _this.inputEl = null;
        _this.onInputKeyDown = function (event) {
            if (event.keyCode === Keys.enter) {
                _this.sendMessage(event);
            }
        };
        _this.sendMessage = function (event) {
            if (_this.props.onMessageSend && _this.inputEl !== null) {
                var messageText = _this.inputEl.value;
                if (messageText) {
                    var message = {
                        author: _this.props.user,
                        text: messageText,
                        timestamp: new Date()
                    };
                    _this.props.onMessageSend(message, event);
                }
            }
            if (_this.inputEl !== null) {
                _this.inputEl.value = null;
            }
        };
        return _this;
    }
    NewMessage.prototype.render = function () {
        var _this = this;
        var localizationService = provideLocalizationService(this);
        var messageInput = (React.createElement("input", { type: "text", className: this.getClassNames(), placeholder: this.props.placeholder, onKeyDown: this.onInputKeyDown, onFocus: function () { return _this.setState({ focused: true }); }, onBlur: function () { return _this.setState({ focused: false }); }, ref: function (el) { return _this.inputEl = el; }, "aria-label": localizationService.toLanguageString(newMessageMessageInput, messages[newMessageMessageInput]) }));
        var sendButton = (React.createElement(Button, { fillMode: "flat", themeColor: "base", className: "k-chat-send", icon: "paper-plane", svgIcon: paperPlaneIcon, onClick: this.sendMessage, "aria-label": localizationService.toLanguageString(newMessageSendButton, messages[newMessageSendButton]), dir: this.props.isDirectionRightToLeft ? 'rtl' : undefined }));
        var toolbarButton = (React.createElement(Button, { fillMode: "flat", themeColor: "base", rounded: null, className: "k-button-toggle", icon: 'more-horizontal', svgIcon: moreHorizontalIcon, onClick: function (event) { return _this.props.onToolbarActionButtonClick(!_this.props.showToolbar, event); }, "aria-label": localizationService.toLanguageString(newMessageOptionsButton, messages[newMessageOptionsButton]) }));
        return (React.createElement(React.Fragment, null,
            React.createElement("span", { className: "k-message-box k-textbox k-input k-input-md k-rounded-md k-input-solid" }, this.props.MessageBox ?
                React.createElement(this.props.MessageBox, { messageInput: messageInput, sendButton: sendButton, toolbarButton: toolbarButton }) :
                React.createElement(React.Fragment, null,
                    messageInput,
                    React.createElement("span", { className: "k-input-suffix" },
                        this.props.toolbar && toolbarButton,
                        sendButton))),
            this.props.showToolbar && React.createElement(Toolbar, null, this.props.toolbar)));
    };
    NewMessage.prototype.focusInput = function () {
        if (this.inputEl !== null) {
            this.inputEl.focus();
        }
    };
    NewMessage.prototype.getClassNames = function () {
        return classNames('k-input-inner', { 'k-focus': this.state.focused });
    };
    return NewMessage;
}(React.Component));
export default NewMessage;
